homeassistant/components/androidtv/config_flow.py

```diff
@@ -0,0 +1,304 @@
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize Android TV config flow."""
+        self._host = None
+        self._unique_id = None
+
+    @callback
+    def _show_setup_form(self, user_input=None, errors=None):
+        """Show the setup form to the user."""
+
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+                    vol.Required(
+                        CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS
+                    ): vol.In(DEVICE_CLASSES),
+                    vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                    vol.Optional(CONF_ADBKEY): str,
+                    vol.Optional(CONF_ADB_SERVER_IP): str,
+                    vol.Required(
+                        CONF_ADB_SERVER_PORT, default=DEFAULT_ADB_SERVER_PORT
+                    ): cv.port,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def _async_check_connection(self, user_input):
+        """Attempt to connect the Android TV."""
+
+        try:
+            aftv = await async_connect_androidtv(self.hass, user_input, timeout=30.0)
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Unknown error connecting with Android TV at %s", self._host
+            )
+            return RESULT_UNKNOWN
+
+        if not aftv:
+            return RESULT_CONN_ERROR
+
+        dev_prop = aftv.device_properties
+        self._unique_id = dev_prop.get("ethmac") or dev_prop.get("wifimac")
+        await aftv.adb_close()
+        return RESULT_SUCCESS
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return self._show_setup_form(user_input)
+
+        errors = {}
+        self._unique_id = None
+        self._host = user_input[CONF_HOST]
+        adb_key = user_input.get(CONF_ADBKEY)
+        adb_server = user_input.get(CONF_ADB_SERVER_IP)
+
+        if adb_key and adb_server:
+            errors["base"] = "key_and_server"
+        elif adb_key:
+            isfile = await self.hass.async_add_executor_job(_is_file, adb_key)
+            if not isfile:
+                errors["base"] = "adbkey_not_file"
+
+        if not errors:
+            ip_address = await self.hass.async_add_executor_job(_get_ip, self._host)
+            if not ip_address:
+                errors["base"] = "invalid_host"
+
+        if errors:
+            return self._show_setup_form(user_input, errors)
+
+        self._async_abort_entries_match({CONF_HOST: self._host})
+        if ip_address != self._host:
+            self._async_abort_entries_match({CONF_HOST: ip_address})
```

Comment:
Is there a reason why we have to evaluate an address other than what the user specifies? I'm assuming that `ip_address` is what the adb device reports?

Reply:
```suggestion
            self._async_abort_entries_match({CONF_HOST: host})
            if ip_address != host:
                self._async_abort_entries_match({CONF_HOST: ip_address})
```